db/crud.go

```diff
@@ -3259,7 +3259,20 @@ func (db *DatabaseCollectionWithUser) CheckProposedVersion(ctx context.Context,
 		// previousVersion didn't match, but proposed version and server CV have matching source, and proposed version is newer
 		return ProposedRev_OK, ""
 	} else {
-		// Conflict, return the current cv.  This may be a false positive conflict if the client has replicated
+		// Temporary (CBG-4466): check the full HLV that's being sent by CBL with proposeChanges messages.
+		// If the current server cv is dominated by the incoming HLV (i.e. the incoming HLV has an entry for the same source
+		// with a version that's greater than or equal to the server's cv), then we can accept the proposed version.
+		proposedHLV, _, err := ExtractHLVFromBlipMessage(proposedHLVString)
+		if err != nil {
+			base.DebugfCtx(ctx, base.KeyCRUD, "CheckProposedVersion for doc %s unable to extract proposedHLV from rev message, treating as conflict: %v", base.UD(docid), err)
```

Comment:
If this happens, this actually seems more fatal than debug, this means that CBL violated the contract/protocol and we would want to know about it. I think we should log as much information as we can here, like `proposedHLVString`